### PR TITLE
Remove trailing whitespace

### PR DESCRIPTION
**Pull request by rainersigwald**
It's been driving me up the wall that half the time I or anyone else edits in an editor that respects `.editorconfig` we wind up with a bunch of irrelevant trailing-whitespace deletions. So let's just do all the deletions!

Find and replace `\s+$` in `{*.cs,*.props,*.targets}`, then reverted `src/Build.OM.UnitTests/Construction/WhiteSpacePreservation_Tests.cs` since it was explicitly working with whitespace.

### CODE REVIEW

File: src/Tasks.UnitTests/RoslynCodeTaskFactory_Tests.cs [hunk 1]
```diff
@@ -17,16 +17,31 @@
...
+using VerifyXunit;
 using Xunit;
 
+using static VerifyXunit.Verifier;
+
 #nullable disable
 
 namespace Microsoft.Build.Tasks.UnitTests
 {
+    [UsesVerify]
```
Comment (rainersigwald) [inline src/Tasks.UnitTests/RoslynCodeTaskFactory_Tests.cs:30]: Alert, alert: actual real changes in this file.

**Comment by JanKrivanek** [comment]
`dotnet format whitespace` on steroids :-) (while so simple)

**Comment by rainersigwald** [comment]
Yeah I did that more at first but it change a bunch of in-the-line spaces and that's not what's been annoying me :)

**Review by JanKrivanek — APPROVED** [review]
Looks good.
I mostly ignored the first commit though

File: src/Tasks.UnitTests/RoslynCodeTaskFactory_Tests.cs [hunk 2]
```diff
@@ -1190,18 +766,10 @@ private void TryLoadTaskBodyAndExpectFailure(string taskBody, string expectedErr
                 taskInfo.CodeType.ShouldBe(expectedCodeType.Value);
             }
 
-            if (expectedSourceCode != null)
+            if (verifySource)
             {
-                NormalizeRuntime(taskInfo.SourceCode)
-                    .ShouldBe(NormalizeRuntime(expectedSourceCode), StringCompareShould.IgnoreLineEndings);
+                Verify(taskInfo.SourceCode, _verifySettings).GetAwaiter().GetResult();
```
Comment (Forgind) [inline src/Tasks.UnitTests/RoslynCodeTaskFactory_Tests.cs:771]: Can you explain for the unenlightened how this is figuring out what it should be? I tend to trust hardcoded 'expected' values, whereas I'm a bit more finnicky about autogenerating them.

Comment (Forgind) [reply src/Tasks.UnitTests/RoslynCodeTaskFactory_Tests.cs:771]: Oh, the files are below. Well, I'm not sure if I like it more or less, but that at least makes me feel more comfortable, so 👍

**Review by Forgind — APPROVED** [review]
Some of these were in Deprecated. I'd kinda expected you'd just touch real code, but I don't really mind.

File: src/Tasks.UnitTests/RoslynCodeTaskFactory_Tests.cs (hunk 2 above)
Comment (rainersigwald) [reply src/Tasks.UnitTests/RoslynCodeTaskFactory_Tests.cs:771]: Yeah, it uses [`Verify`](https://github.com/VerifyTests/Verify) which we started using for similar snapshot testing for `-terminallogger` but I like it.

**Pull request merged**